Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -37,9 +37,7 @@ extension NavigationMapView {
         let nestedList = route.legs.map { (routeLeg: RouteLeg) -> [[CLLocationCoordinate2D]] in
             return routeLeg.steps.map { (routeStep: RouteStep) -> [CLLocationCoordinate2D] in
                 if let routeShape = routeStep.shape {
-                    if !routeShape.coordinates.isEmpty {
-                        return routeShape.coordinates
-                    } else { return [] }
+                    return routeShape.coordinates
```

Comment:
Even `mapbox-directions-swift` parses the route with one more duplicate coordinate in the last step of each route leg, we don't want to drop it here. Because in the `navigationMapView.updateUpcomingRoutePointIndex(routeProgress:)`, we still have to deal with the coordinate of the route step. In this case, we still store them all to be synced with the `routeProgress` that comes in later.